### ISSUE TITLE
LIKWID NvMarkerAPI segfaults on Quadro RTX 5000

### ISSUE THREAD

**Issue by TomTheBear** [RRZE-HPC/likwid#312]
Hi,

I tried this with a simple CUDA kernel.

I have two Quadro RTX 5000 GPUs and use cuda/10.1.243.

It compiles fine with: 
`$ nvcc -DLIKWID_NVMON -L/home/bgursoy/likwid/5.0.1/lib -I/home/bgursoy/likwid/5.0.1/include matrixmult.cu -llikwid`

I run as follows:
```
$ likwid-perfctr -G 0 -W MEM -m ./a.out
--------------------------------------------------------------------------------
CPU name:	Intel(R) Xeon(R) CPU E5-2630 v3 @ 2.40GHz
CPU type:	Intel Xeon Haswell EN/EP/EX processor
CPU clock:	2.39 GHz
NVIDIA PerfWorks API current not supported. Trying CUPTI
Segmentation fault (core dumped)
```

It says here: https://developer.nvidia.com/perfworks that Quadra is a supported platform of PerfWorks however I am getting the warning. It looks like it also doesn’t work with CUPTI. (The code is correct.) 

What would be causing the issue?
Thanks for your time.

Regards,
Buket

_Originally posted by @buketbenek in https://github.com/RRZE-HPC/likwid/issues/311#issuecomment-669194791_

**Comment by TomTheBear** [comment]
The message
`NVIDIA PerfWorks API current not supported. Trying CUPTI`
is not about the capabilities of your GPU but of LIKWID. LIKWID currently supports only event definitions coming from CUPTI but not from PerfWorks. The PerfWorks API is still under development and only partly published. At least the last time I have checked it.

I think the problem is that the function pointer is not properly set up due to the detection of a more recent GPU that supports PerfWorks. The fallback seems not to work. I'll check that.

**Comment by buketbenek** [comment]
Dear Thomas,

Thanks a lot for your reply. Below is the same output for the provided example of likwid.

$ nvcc -I/home/bgursoy/likwid-5.0.1/include -DLIKWID_NVMON triad.cu -L/home/bgursoy/likwid-5.0.1/lib -llikwid
$  
$ ./a.out 
Running without GPU Marker API. Activate GPU Marker API with -m, -G and -W on commandline.
 11.05ms 388.76GB/s 
$ 
$ likwid-perfctr -G 0 -W MEM -m ./a.out
--------------------------------------------------------------------------------
CPU name:	Intel(R) Xeon(R) CPU E5-2630 v3 @ 2.40GHz
CPU type:	Intel Xeon Haswell EN/EP/EX processor
CPU clock:	2.39 GHz
NVIDIA PerfWorks API current not supported. Trying CUPTI
Segmentation fault (core dumped)

I will be waiting from you.

Best Regards,
Buket

**Comment by TomTheBear** [comment]
I was able to reproduce the problem on our machines. I havn't looked into it in detail yet, but it seems the current implementation supports only compute capability < 7.0. For CC >= 7.0, it fails when calling `cuptiDeviceGetNumEventDomains()`

**Comment by buketbenek** [comment]
Dear Thomas,

Thanks a lot for looking into this.

It says here https://docs.nvidia.com/cupti/Cupti/r_main.html#r_event_api : Note: Events APIs from the header cupti_events.h are not supported for devices with compute capability 7.5 and higher.

Since cuptiDeviceGetNumEventDomains() is a function of CUPTI Event API, can it be the reason that CUPTI is failing on Quadros?

Regards,
Buket

**Comment by TomTheBear** [comment]
Hi,

thanks for the link. I knew that the CUPTI Metrics API is deprecated and replaced by PerfWorks Metric API but that CUPTI Event API is outdated is new to me. So, yes, that's probably the reason.

So, either I do a "hard exit" when running LIKWID with a Nvidia GPU with CC >= 7.5 or I update the interface to work independent of the CC.

Best,
Thomas

**Comment by buketbenek** [comment]
Dear Thomas,

The second option would be great for us of course. :)

I will keep an eye on future likwid updates.

Thanks again.
Regards,
Buket

**Comment by TomTheBear** [comment]
The current master branch should work for you. I try to release a new. version as soon as possible.

**Comment by TomTheBear** [comment]
Fixed in 5.1.0
